%YAML:1.0

camera_model: pinhole
distortion_model: "radtan"
image_width: 1024
image_height: 768


T_cam0_imu: !!opencv-matrix
  rows: 4
  cols: 4
  dt: d
  data: [0.9999180, 0.0127351, 0.0013603, -0.05,
        0.0014684, -0.0084771, -0.9999630, 0.0,
        -0.0127231, 0.9998830, -0.0084951, 0.001,
        0.0, 0.0, 0.0, 1.0]

# T_cam0_imu: !!opencv-matrix
#   rows: 4
#   cols: 4
#   dt: d
#   data: [0.99991796, 0.001468326707, -0.01272309666, 0.05000862109,
#         0.01273510271, -0.008477095676, 0.9998829423, -0.0003631278069,
#         0.001360373127, -0.9999629814, -0.008495104604, 0.00007651376096,
#         0.0, 0.0, 0.0, 1.0]

cam0_distortion_coeffs: !!opencv-matrix    
  rows: 1
  cols: 4
  dt: d
  data: [0, 0, 0, 0]

cam0_intrinsics: !!opencv-matrix
  rows: 1
  cols: 4
  dt: d
  data: [665.107510106, 665.107510106, 511.5, 383.5]


T_cam1_imu: !!opencv-matrix
  rows: 4
  cols: 4
  dt: d
  data: [0.9999180, 0.0127351, 0.0013603, 0.05,
        0.0014684, -0.0084771, -0.9999630, 0.0,
        -0.0127231, 0.9998830, -0.0084951, -0.001,
        0.0, 0.0, 0.0, 1.0]

# T_cam1_imu: !!opencv-matrix
#   rows: 4
#   cols: 4
#   dt: d
#   data: [0.99991796, 0.001468326707, -0.01272309666, -0.05000862109,
#         0.01273510271, -0.008477095676, 0.9998829423, 0.0003631278069,
#         0.001360373127, -0.9999629814, -0.008495104604, -0.00007651376096,
#         0.0, 0.0, 0.0, 1.0]
        
T_cn_cnm1: !!opencv-matrix
  rows: 4
  cols: 4
  dt: d
  data: [1.0, 0.0, 0.0, -0.1,
        0.0, 1.0, 0.0, 0.000,
        0.0, 0.0, 1.0, 0.002,
        0.0, 0.0, 0.0, 1.0]

cam1_distortion_coeffs: !!opencv-matrix   
  rows: 1
  cols: 4
  dt: d
  data: [0, 0, 0, 0]

cam1_intrinsics: !!opencv-matrix
  rows: 1
  cols: 4
  dt: d
  data: [665.107510106, 665.107510106, 511.5, 383.5]


#================================ ROS Settings ================================

imu_topic: "/blackbird/imu"
motors_topic: "/blackbird/rotor_rpm"
cam0_topic: "/uav/camera/Camera_Left_Gray/image_rect"
cam1_topic: "/uav/camera/Camera_Right_Gray/image_rect"

#================================= Frontend Settings ===========================

grid_row: 4
grid_col: 5
grid_min_feature_num: 3
grid_max_feature_num: 4
pyramid_levels: 3
patch_size: 3
fast_threshold: 10
klt_max_iteration: 30
klt_track_precision: 0.01

